src/types/browse.d.ts

```diff
@@ -77,40 +78,52 @@ declare class Browse {
 /** *********
  * Browse results returned from server
  ********** */
-export interface BrowseResponse<ResponseType> extends Record<string, any> {
-  request?: Partial<BrowseRequestType>;
-  response?: Partial<ResponseType>;
-  result_id?: string;
+export interface BrowseResponse<ResponseType, OmittedRequestFields extends keyof BrowseRequestType>
+  extends Record<string, any> {
+
+  request: Omit<BrowseRequestType, OmittedRequestFields>;
+  response: ResponseType;
+  result_id: string;
   ad_based?: boolean;
 }
 
 export type GetBrowseResultsResponse =
-  BrowseResponse<GetBrowseResultsResponseData>;
+  BrowseResponse<GetBrowseResultsResponseData, 'facet_name'>;
 export type GetBrowseResultsForItemIdsResponse =
-  BrowseResponse<GetBrowseResultsResponseData>;
+  BrowseResponse<GetBrowseResultsResponseData, 'facet_name'>;
 export type GetBrowseGroupsResponse = BrowseResponse<
   Pick<
     GetBrowseResultsResponseData,
     'result_sources' | 'groups' | 'refined_content'
-  >
+  >,
+  'browse_filter_name' | 'browse_filter_value' | 'searchandized_items' | 'facet_name'
 >;
 export type GetBrowseFacetsResponse = BrowseResponse<
-  Pick<GetBrowseResultsResponseData, 'facets' | 'total_num_results'>
+  GetBrowseFacetsResultsResponseData,
```

Comment:
The facets response actually returns a lighter version of the facet data and therefore I made a separate interface for it.